pyproject.toml

```diff
@@ -230,6 +230,21 @@ env.OPTS = """
 --cov-context=test
 """
 
+[tool.poe.tasks.check-temp-files]
+help = "Run each test module one by one and check for leftover temp files"
+shell = """
+setopt nullglob
+for file in test/**/*.py; do
+  print Temp files created by $file && poe test $file &>/dev/null
+  tempfiles=(/tmp/**/tmp* /tmp/beets/**/*)
+  if (( $#tempfiles )); then
+    print -l $'\t'$^tempfiles
+    rm -r --interactive=never $tempfiles &>/dev/null
+  fi
+done
+"""
+interpreter = "zsh"
+
```

Comment:
I think the downside here is how long it takes to run - I have done it now and it took longer than 4 minutes.

In any case, there's probably little value for it once these issues are fixed - but it's good to have it around in case this issue comes up again.

Reply:
That's interesting, why does it take longer to run than the normal test suite?

I suppose it doesn't actually matter, once we switch to pytest, it'll handle all of the temp files and directories and fix issues.

Additionally this issue is more of a nice to have and to stop any possible dependence between test runs because the OS should clear out temp files, at the very least every reboot. That's what the temp files are for.

Reply:
> That's interesting, why does it take longer to run than the normal test suite?

Because it runs `pytest` for each of the test files :)

Reply:
Do you mean that it runs the entire test suite for each of the files?

Reply:
Nope, each invocation runs tests found in each of these files 

```
$ centralize-test-setup print -l test/**/*.py                                                                                                  beets 2.0.0 │ 🐍 3.8 🐶
test/__init__.py
test/plugins/__init__.py
test/plugins/lyrics_download_samples.py
test/plugins/test_acousticbrainz.py
test/plugins/test_advancedrewrite.py
test/plugins/test_albumtypes.py
test/plugins/test_art.py
test/plugins/test_aura.py
test/plugins/test_bareasc.py
test/plugins/test_beatport.py
```